base/interactiveutil.jl

```diff
@@ -254,8 +254,14 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit] && ex0.head == :macrocall
+        # special case @edit @time 1+2 to edit the macro
+        is_macro = true
```

Comment:
The special case is needed because when the `expand` function called above expands the macro/insert its code into the expression, and hence the `edit` call will call edit on the expanded code. The special casing makes `which`, `less` and `edit` macro call the macrocall instead of the code the macro expands to. This enables the user to write: `@edit @time 1+1`, this will go to the `@time` macro's definition.